### PR TITLE
fix: [M3-7718] - Stale assigned Firewall data displaying on Linode and NodeBalancer details pages

### PR DESCRIPTION
**Pull request by mjac0bs**
## Description 📝
`useAllFirewallDevicesQuery` was not always being invalidated when modifying a Firewall.

As a result, without refreshing the page, adding/deleting rules or enabling/disabling a Firewall with assigned device(s) will not show the Firewall's most accurate status or rules on the NodeBalancer Details > Settings  or Linode Details > Network tab if the query for that Firewall has already been cached because the page was previously visited.

## Changes  🔄
- Added query invalidation on submission of the Firewall Dialog for enabling and disabling a Firewall.
- Added query invalidation on submission of updated Firewall rules.
- Cleaned up naming of query keys for consistency (capitalization; made query key plural for `nodebalancers`).

## Target release date 🗓️
6/10

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/114685994/d78b20fc-3cfd-4b1e-a103-84ebedcdf381" /> | <video src="https://github.com/linode/manager/assets/114685994/ab8fffb5-1f40-4a85-bd1e-f28cd2977bc2" /> |

### Prerequisites
(How to setup test environment)
- Have/create a device (Linode and/or NodeBalancer) on your account.

### Reproduction steps
(How to reproduce the issue, if applicable)
1. Go to https://cloud.linode.com/firewalls and create a new Firewall.
2. Attach your device to that Firewall.
3. Click on the label of the device to navigate to the Details page.
4. Go to the Linode Details > Network tab or NodeBalancer Details > Settings tab. Observe under the Firewalls section that the Firewall from step 2 is listed and there are "No rules" and it is Enabled.
5. Click on the Firewall label to navigate to the Firewall's details page and add a new inbound or outbound rule. (Can be a preset rule; it's fastest.) Save the changes. 
6. Switch the Firewall's Status from Enabled to Disabled via the Firewall landing page.
7. Navigate back to the device with the Firewall, either via nav bar or via the Firewall details page > device tab.
8. Go back to the Network or Settings tab where the Firewalls table is shown for that device.
9. Observe the bug: the Firewall listed in the table will not have the newly added rule and its status will not have updated.
10. Refresh the page.
111. Observe the rule now shows up. (e.g. "1 inbound rule" or "1 outbound rule") and the status has changed.

### Verification steps
(How to verify changes)
- Check out this PR.
- Repeat the steps above to modify the Firewall's rules and status. 
- Confirm that no page refresh is needed to see the device's accurate Firewall data due to RQ correctly invalidating as expected when the Firewall is modified, and then refetching once the user navigates to the Firewalls table on the Linode Network tab or NodeBalancer Settings tab.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Firewalls/FirewallLanding/FirewallDialog.tsx [hunk 1]
```diff
@@ -66,17 +66,19 @@ export const FirewallDialog = React.memo((props: Props) => {
 
   const onSubmit = async () => {
     await requestMap[mode]();
+    // Invalidate Firewalls assigned to NodeBalancers and Linodes when Firewall is enabled, disabled, or deleted.
+    // eslint-disable-next-line no-unused-expressions
+    devices?.forEach((device) => {
```
Comment (mjac0bs) [inline packages/manager/src/features/Firewalls/FirewallLanding/FirewallDialog.tsx:71]: Previously, we were only invalidating the Firewalls assigned to a device when deleting the Firewall. Updating the status of the Firewall (e.g. enabling it or disabling it) should be reason to invalidate the Firewall too.

File: packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRulesLanding.tsx [hunk 2]
```diff
@@ -193,6 +201,19 @@ export const FirewallRulesLanding = React.memo((props: Props) => {
     updateFirewallRules(finalRules)
       .then((_rules) => {
         setSubmitting(false);
+        // Invalidate Firewalls assigned to NodeBalancers and Linodes.
```
Comment (mjac0bs) [inline packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRulesLanding.tsx:204]: No invalidation was happening on Firewall rule updates, which was the initial issue identified in the ticket.

File: packages/manager/src/features/Firewalls/FirewallDetail/Devices/RemoveDeviceDialog.tsx
Comment (mjac0bs) [inline packages/manager/src/features/Firewalls/FirewallDetail/Devices/RemoveDeviceDialog.tsx:1]: Just some linting and renaming in this file. The query key for nodebalancers is plural, so updated the name of the import accordingly. This is done in a few other files, too.

**Comment by mjac0bs** [comment]
The CI test failure on `linode-config.spec.ts` is unrelated to this PR and it passes locally.

image

**Review by jaalah-akamai — APPROVED** [review]

**Review by bnussman-akamai — APPROVED** [review]
Thank you for the fixes! 🧰

**Pull request merged**